### PR TITLE
Feat/jupiter

### PR DESCRIPTION
**Pull request by bryzettler**
### Pulls in all tokens from your visible tokens list and shows swaps based on that. 

jupiter

### CODE REVIEW

**Comment by bryzettler** [comment]
@matthewcarlreetz not that I know of. This was just to add Jupiter as the main means for the SwapsScreen. What needs to be done for location assertion?

**Comment by matthewcarlreetz** [comment]
Nothing necessarily. I was just thinking we could allow them to pay the re-assert fee with IOT, MOBILE, or HNT. Right now it will only burn HNT (if needed).

File: src/features/swaps/SwapScreen.tsx [hunk 1]
```diff
@@ -123,19 +155,16 @@ const SwapScreen = () => {
...
+    if (inputMint.equals(HNT_MINT) && (hntBalance || new BN(0)).lt(new BN(1))) {
       return true
     }
 
     return (
-      !youPayMint.equals(HNT_MINT) &&
-      !(price && price > 0) &&
-      youPayTokenAmount > 0
+      !inputMint.equals(HNT_MINT) &&
+      !(routes?.outAmount && Number(routes?.outAmount) > 0) &&
```
Comment (ChewingGlass) [inline src/features/swaps/SwapScreen.tsx:164]: Can this blowup with big numbers?

File: src/features/swaps/SwapScreen.tsx [hunk 2]
```diff
@@ -206,6 +232,44 @@ const SwapScreen = () => {
...
+      if (
+        validOutputMints &&
+        !validOutputMints?.includes(outputMint?.toBase58() || '')
+      ) {
+        setOutputMint(new PublicKey(validOutputMints[0]))
+      }
+    }
+  }, [inputMint, outputMint, validOutputMints])
+
+  useEffect(() => {
```
Comment (ChewingGlass) [inline src/features/swaps/SwapScreen.tsx:247]: Do we want to useAsync here and catch/display errors?

**Review by ChewingGlass — CHANGES_REQUESTED** [review]

**Pull request closed without merging**